### PR TITLE
CIRC-345 - fixed defect requiring RequestExpirationDate in the instance-level Request

### PR DESCRIPTION
**Pull request by mtraneis**
## Purpose
The current instance-level request code assumes that RequestExpirationDate is there in the request body so it causes a NullPointerException when it is not present. 

## Approach
Check for the property requestExpirationDate being null. If not null add it to the converted Request object.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/api/requests/InstanceRequestsAPICreationTests.java [hunk 1]
```diff
@@ -108,6 +108,41 @@ public void canCreateATitleLevelRequestForOneAvailableCopy()
...
+    CompletableFuture<Response> postCompleted = new CompletableFuture<>();
+
+    client.post(InterfaceUrls.requestsUrl("/instances"), requestBody,
+      ResponseHandler.any(postCompleted));
+
+    Response postResponse = postCompleted.get(10, TimeUnit.SECONDS);
+
+    JsonObject representation = postResponse.getJson();
+    validateInstanceRequestResponse(representation,
+      pickupServicePointId,
```
Comment (mreno-EBSCO) [inline src/test/java/api/requests/InstanceRequestsAPICreationTests.java]: Should we be validating that `requestExpirationDate` is not set here, as opposed to just validate we don't throw a `NullPointerException`?

File: src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java [hunk 2]
```diff
@@ -216,9 +216,11 @@ private void createInstanceLevelRequests(RoutingContext routingContext) {
           requestBody.put("requesterId", requestByInstanceIdRequest.getRequesterId().toString());
           requestBody.put("pickupServicePointId", requestByInstanceIdRequest.getPickupServicePointId().toString());
           requestBody.put("fulfilmentPreference", defaultFulfilmentPreference);
-          requestBody.put("requestExpirationDate",
-            requestByInstanceIdRequest.getRequestExpirationDate().toString(ISODateTimeFormat.dateTime()));
           requestBody.put("requestType", reqType.name());
+          if (requestByInstanceIdRequest.getRequestExpirationDate() != null) {
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java]: This check can be replaced by using the `JsonPropertyWriter`. Only a thought for another day, does not hold up merging.

File: src/test/java/api/requests/InstanceRequestsAPICreationTests.java (hunk 1 above)
Comment (mtraneis) [reply src/test/java/api/requests/InstanceRequestsAPICreationTests.java]: Yeah, we could do something like that here. I'll put it in the next commit.

File: src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java (hunk 2 above)
Comment (mtraneis) [reply src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java]: I'll keep this option in mind for later refactoring.

**Pull request merged**